docs/ref/methods_and_combinators.rst

```diff
@@ -571,6 +571,29 @@ Parser combinators
       For earlier versions, see :meth:`Parser.tag` for a way of labelling parsed
       components and producing dictionaries.
 
+.. function:: many_until(parser, end)
+
+   Creates a parser that expects :arg:`parser` 0 or more times until :arg:`end` is
+   found.
+
+   .. code-block:: python
+
+      >>> many_until(string("a"), string("abc")).parse("aaabc")
+      ['a', 'a', 'abc']
+
+   Note that even though ``string("a")`` and ``string("abc")`` overlap, ``end``
+   parser is tried first after each match of ``parser``. Otherwise the example
+   above would fail with ``['a', 'a', 'a', ParseError]``.
```

Comment:
I think this would be clearer as something like "would fail with ParseError" - we don't collect ParseError values in lists and these docs might imply that.